src/core/etl/src/Flow/ETL/PHP/Type/Logical/DateTimeType.php

```diff
@@ -4,102 +4,76 @@
+    public function cast(mixed $value) : \DateTimeInterface
     {
-        if ($type instanceof self) {
-            return true;
+        if ($value instanceof \DateTimeImmutable) {
+            return $value;
         }
 
-        if ($type instanceof NullType) {
-            return true;
+        if ($value instanceof \DOMElement) {
+            $value = $value->nodeValue;
         }
 
-        if ($type instanceof DateType) {
-            return true;
+        if ($value instanceof \DateTime) {
+            return \DateTimeImmutable::createFromMutable($value);
         }
 
-        return false;
-    }
-
-    public function isCompatible(Type $type) : bool
-    {
-        if (!$this->nullable && $type->nullable()) {
-            return false;
-        }
+        try {
+            if (\is_string($value)) {
+                return new \DateTimeImmutable($value);
+            }
 
-        return $this->isEqual($type);
-    }
+            if (\is_numeric($value)) {
+                return new \DateTimeImmutable('@' . $value);
+            }
 
-    public function isEqual(Type $type) : bool
-    {
-        return $type instanceof self;
-    }
+            if (\is_bool($value)) {
+                /* @phpstan-ignore-next-line */
+                return new \DateTimeImmutable('@' . $value);
+            }
 
-    public function isSame(Type $type) : bool
-    {
-        return $this->isEqual($type) && $this->nullable() === $type->nullable();
-    }
+            if ($value instanceof \DateInterval) {
+                return (new \DateTimeImmutable('@0'))->add($value);
 
-    public function isValid(mixed $value) : bool
-    {
-        if ($this->nullable && $value === null) {
-            return true;
+            }
+        } catch (\Throwable) {
+            throw new CastingException($value, $this);
         }
 
-        return $value instanceof \DateTimeInterface;
-    }
-
-    public function makeNullable(bool $nullable) : self
-    {
-        return new self($nullable);
+        throw new CastingException($value, $this);
     }
 
-    public function merge(Type $type) : self
+    public function isValid(mixed $value) : bool
     {
-        if ($type instanceof NullType) {
-            return $this->makeNullable(true);
-        }
-
-        if (!$type instanceof self) {
-            throw new InvalidArgumentException('Cannot merge different types, ' . $this->toString() . ' and ' . $type->toString());
-        }
-
-        return new self($this->nullable || $type->nullable());
+        return $value instanceof \DateTimeInterface;
     }
 
     public function normalize() : array
     {
         return [
             'type' => 'datetime',
-            'nullable' => $this->nullable,
         ];
     }
 
-    public function nullable() : bool
-    {
-        return $this->nullable;
-    }
-
     public function toString() : string
     {
-        return ($this->nullable ? '?' : '') . 'datetime';
+        return 'datetime';
```

Comment:
I don't know the context where this was used but shouldn't the `type_optional` recognize it?

Reply:
Previously all types has `$nullable` property that why `date_type(true)` was returning `?datetime`. Now to achieve the same goal we need to do `type_optional(type_date())`